index.bs

```diff
@@ -2043,9 +2101,16 @@ attributes.
         specified [[#attachment]].
 
     :   <dfn>requireResidentKey</dfn>
-    ::  This member describes the [=[RP]=]'s requirements regarding [=resident credentials=].
-        If the parameter is set to [TRUE], the authenticator MUST create a
-        [=client-side-resident public key credential source=] when creating a [=public key credential=].
+    ::  This member is retained for backwards compatibility but is deprecated in favour of providing *residentKey*.
+        It will be ignored if *residentKey* is supplied and understood by the user-agent. When provided, it describes the 
+        [=[RP]=]'s requirements regarding [=resident credentials=]. If the parameter is set to [TRUE], the authenticator MUST create 
+        a [=client-side-resident public key credential source=] when creating a [=public key credential=].
+
+    :   <dfn>residentKey</dfn>
+    ::  This member supercedes the use of *requireResidentKey* to describes the [=[RP]=]'s requirements regarding 
+        [=resident credentials=]. Only one of *residentKey* or *requireResidentKey* should be present in 
```

Comment:
Should we say that only one should be present? Providing both means being backwards compatible with level 1 of the spec so I might almost go the other way: &ldquo;if {{residentKey}} is set to `required` then {{requireResidentKey}} should be `true`, else it should be omitted&rdquo;. Don't feel strongly either way however.

Reply:
I'm going to leave this for now. I feel it introduces ambiguity as to what `requireResidentKey` should be if supplied alongside `residentKey`=`preferred`. I feel it's better just to instruct the client to apply `residentKey` only (and ignore `requireResidentKey`) if it understand the L2 definition.

Reply:
If, as an RP, I want to create a resident key, I could set `residentKey` to `required`, but that doesn't work for any browser only implementing Webauthn level one. So I could set `requireResidentKey` too to cover both, but this says that only one of them should be present, so I remove `residentKey`.

I think that's ok, but it does mean that we won't see the transition that we might otherwise.

Reply:
Ok - I will change the wording to be more accepting of both being present.